e3sm_diags/driver/utils/regrid.py

```diff
@@ -0,0 +1,635 @@
+    -------
+    xr.Dataset
+        The dataest with the subsetted variable.
+
+    Notes
+    -----
+    Replaces `e3sm_diags.utils.general.select_region`.
+    """
+    specs = REGION_SPECS[region]
+
+    lat, lon = specs.get("lat"), specs.get("lon")  # type: ignore
+
+    if lat is not None:
+        lat_dim = xc.get_dim_keys(ds[var_key], axis="Y")
+        ds = ds.sel({f"{lat_dim}": slice(*lat)})
+
+    if lon is not None:
+        lon_dim = xc.get_dim_keys(ds[var_key], axis="X")
+        ds = ds.sel({f"{lon_dim}": slice(*lon)})
+
+    return ds
+
+
+def _subset_on_arm_coord(ds: xr.Dataset, var_key: str, arm_site: str):
+    """Subset a variable in the dataset on the specified ARM site coordinate.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        The dataset.
+    var_key : str
+        The variable to subset.
+    arm_site : str
+        The ARM site.
+
+    Notes
+    -----
+    Replaces `e3sm_diags.utils.general.select_point`.
+    """
+    # TODO: Refactor this method
+    pass
+
+
+def align_grids_to_lower_res(
+    ds_a: xr.Dataset,
+    ds_b: xr.Dataset,
+    var_key: str,
+    tool: REGRID_TOOLS,
+    method: str,
+) -> Tuple[xr.DataArray, xr.DataArray]:
+    """Align the grids of two Dataset using the lower resolution of the two.
+
+    A variable has a lower resolution if it has less latitude coordinates,
+    and vice versa. If both resolutions are the same, no regridding will happen.
+
+    Parameters
+    ----------
+    ds_a : xr.Dataset
+        The first Dataset containing ``var_key``.
+    ds_b : xr.Dataset
+        The second Dataset containing ``var_key``.
+    var_key : str
+        The key of the variable in both datasets to regrid.
+    tool : {"esmf", "xesmf", "regrid2"}
+        The regridding tool to use. Note, "esmf" is accepted for backwards
+        compatibility with e3sm_diags and is simply updated to "xesmf".
+    method : str
+        The regridding method to use. Refer to [1]_ for more information on
+        these options.
+
+        esmf/xesmf options:
+          - "bilinear"
+          - "conservative"
+          - "conservative_normed"
+          - "patch"
+          - "nearest_s2d"
+          - "nearest_d2s"
+
+        regrid2 options:
+          - "conservative"
+
+    Returns
+    -------
+    Tuple[xr.DataArray, xr.DataArray]
+        A tuple of both DataArrays regridded to the lower resolution of the two.
+
+    Notes
+    -----
+    Replaces `e3sm_diags.driver.utils.general.regrid_to_lower_res`.
+
+    References
+    ----------
+    .. [1] https://xcdat.readthedocs.io/en/stable/generated/xarray.Dataset.regridder.horizontal.html
+    """
+    # TODO: Accept "esmf" as `tool` value for now because `CoreParameter`
+    # defines `self.regrid_tool="esmf"` by default and
+    # `e3sm_diags.driver.utils.general.regrid_to_lower_res()` accepts "esmf".
+    # Once this function is deprecated, we can remove "esmf" as an option here
+    # and update `CoreParameter.regrid_tool` to "xesmf"`.
```

Comment:
Does this `TODO` still apply?

Reply:
Yes, this will happen once all sets are refactored and no longer use `general.regrid_to_lower_res()`.

Reply:
Moved to #734.